orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh

```diff
@@ -56,19 +58,26 @@ for existing in "${existingusers[@]}";
 do
   hasharr[$existing]="exist"
 done
-number_of_dirs_created=0
+number_of_dirs=0
+newpaths=""
 for user in "${users[@]}";
 do
   if [[ ${hasharr[$user]} != "exist" ]]; then
-     echo "Trying to create /user/$user directory"
-     (hdfs dfs -mkdir /user/$user) && (hdfs dfs -chown $user /user/$user)
-     if [[ $? -eq 0 ]]; then
-       echo "Home directory created for $user"
-       echo "$user" >> $EXISTING_USERS_FILE
-       number_of_dirs_created=$((number_of_dirs_created + 1))
-     else
-       echo "Directory creation or chown attempt for $user failed. Directory already exists?"
-     fi
+     echo "Will attempt to create HDFS home dir for $user"
+     newpaths+="$PATH_PREFIX/$user "
   fi
 done
-echo "End time of script $0, PID $$: `date`. Number of HDFS home dirs created $number_of_dirs_created."
+if [ -z "$newpaths" ]; then
+  echo "No home dirs to create in HDFS"
+else
+  hdfs dfs -mkdir -p $newpaths
```

Comment:
will it handle hundreds of arguments? this should be tested on mow env before merged

Reply:
Yes tested on mow-dev